torch/_custom_op/autograd.py

```diff
@@ -56,6 +56,37 @@ def kernel(*args, **kwargs):
     return kernel
 
 
+def mark_non_differentiable(ctx, output, output_differentiability):
+    # Output types are restricted to be:
+    # - Tensor
+    # - Tensor[]
+    # - int, bool, Scalar, float
```

Comment:
If the user has lied in their schema they will get some funky pybind errors when trying to return e.g. a None for an int. So yeah, we can assume this because otherwise the schema is wrong

Reply:
I'm going to check this in the next PR up in the stack, which validates the schema.

Reply:
Sounds good!